src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const spr = `${separator}`;
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let e = this[i];
+
+      if (e === null || e === undefined) {
+        e = '';
+      }
+
+      joinedString += spr + e;
+    }
+
+    return joinedString.slice(spr.length);
```

Comment:
is it allowed to use any method in this task?

Reply:
my logic said that's a method of String so I used it)